src/Unleash.php

```diff
@@ -74,31 +74,33 @@ public function isFeatureEnabled(string $name, ...$args): bool
         $strategies = Arr::get($feature, 'strategies', []);
         $allStrategies = $this->config->get('unleash.strategies', []);
 
+        if (count($strategies) == 0) {
+            return $isEnabled;
+        }
+
         foreach ($strategies as $strategyData) {
             $className = $strategyData['name'];
 
-            if (!array_key_exists($className, $allStrategies)) {
-                return false;
-            }
-
-            if (is_callable($allStrategies[$className])) {
-                $strategy = $allStrategies[$className]();
-            } else {
-                $strategy = new $allStrategies[$className];
-            }
-
-            if (!$strategy instanceof Strategy && !$strategy instanceof DynamicStrategy) {
-                throw new \Exception("${$className} does not implement base Strategy/DynamicStrategy.");
-            }
-
-            $params = Arr::get($strategyData, 'parameters', []);
-
-            if (!$strategy->isEnabled($params, $this->request, ...$args)) {
-                return false;
+            if (array_key_exists($className, $allStrategies)) {
```

Comment:
I have refactored to remove the nested `if`.

I think it depends how we want to handle the behavior in a scenario where a multi-strategy feature has been configured, but one of those strategies has not been mapped in the `unleash.strategies` configmap.

For example, when on a feature, the 1st strategy isn't within the configmap, but a second strategy _is_ implemented and evaluates to `true`, we would always failback to `false` because we couldn't run the 1st strategy.

If we return `false` because `array_key_exists` doesn't find the strategy name in the `allStrategies`, then we would never hit any of the other strategies which could potentially resolve to `true`. Is this the behavior we expect?

Await your feedback and thoughts

Reply:
Personally I feel like if a strategy has been configured but there's nothing to handle it, it should default to `false`?

Maybe if the app is in debug mode, it could throw an exception as this might be something you want to know about?

Reply:
100%, so if a strategy has been configured in Unleash but no handler present in the mapping, it would follow the currently proposed behavior (which is to skip that strategy and evaluate any others), returning false if no strategies are matched.